pipeline/store/accessor.go

```diff
@@ -0,0 +1,177 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/streamingfast/dstore"
+	"github.com/streamingfast/substreams/block"
+	pbsubstreams "github.com/streamingfast/substreams/pb/sf/substreams/v1"
+	"github.com/streamingfast/substreams/state"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var ErrStoreNotFound = errors.New("store not found")
+
+type Accessor struct {
+	baseStore           dstore.Store
+	storeMap            map[string]*state.Store
+	SaveInterval        uint64
+	maxBoundaryBlockNum uint64
+	nextSaveBoundary    uint64
+	logger              *zap.Logger
+}
+
+func NewAccessor(saveInterval, minBoundaryBlock, maxBoundaryBlockNum uint64, baseStore dstore.Store, logger *zap.Logger) *Accessor {
+	s := &Accessor{
+		storeMap:            map[string]*state.Store{},
+		SaveInterval:        saveInterval,
+		maxBoundaryBlockNum: maxBoundaryBlockNum,
+		baseStore:           baseStore,
+		logger:              logger,
+	}
+
+	s.nextSaveBoundary = s.computeBoundaryBlock(minBoundaryBlock)
+	return s
+}
+
+func (s *Accessor) AddStore(hash string, storeModule *pbsubstreams.Module) (err error) {
+	store, err := state.NewStore(
+		storeModule.Name,
+		s.SaveInterval,
+		storeModule.InitialBlock,
+		// gets thme for store
+		hash,
+		storeModule.GetKindStore().UpdatePolicy,
+		storeModule.GetKindStore().ValueType,
+		s.baseStore,
+		s.logger,
+	)
```

Comment:
is it the Accessor's responsibility to *create* the store? that can be a function in our pipelines or elsewhere, that creates the store, and then loads it in a map. But for this module to know much about what to get from withiin a pusubstreams.Module.. goes too far imo.